extras/Build/CMake/JUCEUtils.cmake

```
@@ -1337,6 +1347,40 @@ function(_juce_set_plugin_target_properties shared_code_target kind)
         endif()
 
         _juce_copy_after_build(${shared_code_target} ${target_name} "${output_path}" JUCE_VST3_COPY_DIR)
+    elseif(kind STREQUAL "LV2")
+        set_target_properties(${target_name} PROPERTIES
+            BUNDLE_EXTENSION lv2
+            PREFIX ""
+            SUFFIX .lv2
+            BUNDLE TRUE
+            XCODE_ATTRIBUTE_WRAPPER_EXTENSION lv2
+            XCODE_ATTRIBUTE_LIBRARY_STYLE Bundle
+            XCODE_ATTRIBUTE_GENERATE_PKGINFO_FILE YES)
+
+        _juce_create_windows_package(${shared_code_target} ${target_name} lv2 "" x86-win x86_64-win)
+
+        set(output_path "${products_folder}/${product_name}.lv2")
+
+        if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+            # On linux we assume that the output arch is the same as the that of the host platform
+            set(is_platform_x64 $<EQUAL:${CMAKE_SIZEOF_VOID_P},8>)
+            set(arch_string $<IF:${is_platform_x64},x86_64,i386>)
+
+            set_target_properties(${target_name} PROPERTIES
+                SUFFIX .so
+                LIBRARY_OUTPUT_DIRECTORY "${output_path}")
+        endif()
+
+        # generate .ttl files
+        add_executable(lv2_ttl_generator ${JUCE_SOURCE_DIR}/extras/Build/lv2_ttl_generator/lv2_ttl_generator.c)
+        target_link_libraries(lv2_ttl_generator dl)
```

Can I instead suggest:

```suggestion
        target_link_libraries(lv2_ttl_generator ${CMAKE_DL_LIBS})
```

As it should be more portable than just `dl`.

---

Update: I believe you should also add:

```cmake
set_source_files_properties(${JUCE_SOURCE_DIR}/extras/Build/lv2_ttl_generator/lv2_ttl_generator.c PROPERTIES
    LANGUAGE CXX)
```

As it will ensure `lv2_ttl_generator` is built and linked as C++, not C.

---

This change LGTM. I'll test it out and then commit.